### PR TITLE
feature - ui list screen

### PR DESCRIPTION
**Pull request by at-duytran**
I'm working with UI

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/List/FoodViewController.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        super.viewDidLoad()
+    }
+
+    @IBAction private func goingDetailTouchUpInside(_ sender: Any) {
+        let vc = DetailViewController()
+        navigationController?.pushViewController(vc, animated: true)
+    }
+}
+//
+//extension FoodViewController: UITableViewDataSource {
```
Comment (at-daiho) [inline MyApp/View/Controllers/List/FoodViewController.swift]: WARNING - There should be a space after //

File: MyApp/View/Controllers/List/FoodViewController.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+//    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+//        <#code#>
+//    }
+//
+//    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+//        <#code#>
+//    }
+//}
+//
+//extension FoodViewController: UITableViewDelegate {
```
Comment (at-daiho) [inline MyApp/View/Controllers/List/FoodViewController.swift]: WARNING - There should be a space after //

**Comment by at-daiho** [comment]
Total linter violations in pull request: 2
Serious: 0
Warnings: 2

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/List/Cell/ItemListTableCell.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
+
+    func updateView() {
+        guard let viewModel = viewModel else { return }
+        itemBackgroundImage.image = UIImage(named: "\(viewModel.imageName)")
+        itemNameLabel.text = viewModel.name
+        if viewModel.detail == "" {
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/List/Cell/ItemListTableCell.swift]: isEmpty

File: MyApp/View/Controllers/List/DrinkViewController.swift [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        tableView.delegate = self
+        tableView.register(ItemListTableCell.self)
+
+        viewModel.getProductDetail { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success:
+                this.tableView.reloadData()
+            case .failure(let error):
+                print(error)
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/List/DrinkViewController.swift]: alert?

File: MyApp/View/Controllers/List/FavoriteViewController.swift [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        tableView.delegate = self
+        tableView.register(ItemListTableCell.self)
+
+        viewModel.getProductDetail { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success:
+                this.tableView.reloadData()
+            case .failure(let error):
+                print(error)
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/List/FavoriteViewController.swift]: alert

File: MyApp/View/Controllers/List/FoodViewController.swift [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    fileprivate var viewModel = FoodViewModel() {
+        didSet {
+            tableView.reloadData()
+        }
+    }
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    fileprivate let heightForRow: CGFloat = 200
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/List/FoodViewController.swift]: Đổi tên em êy 👯‍♂️ `rowHeight` là đc rồi :)

File: MyApp/View/Controllers/List/FoodViewController.swift [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        tableView.delegate = self
+        tableView.register(ItemListTableCell.self)
+
+        viewModel.getProductDetail { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success:
+                this.tableView.reloadData()
+            case .failure(let error):
+                print(error)
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/List/FoodViewController.swift]: show lên :)

File: MyApp/View/Controllers/List/FoodViewController.swift [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            }
+        }
+    }
+
+    @IBAction private func goingDetailTouchUpInside(_ sender: Any) {
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/List/FoodViewController.swift]: DetailButtonTuch....

File: MyApp/ViewModel/List/DrinkViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    func getProductDetail(_ completion: @escaping GetProductCompletion) {
+        Api.Item.query { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success(let product):
+                this.items = product as? [Item]
+                completion(.success)
+            case .failure(let error):
+                completion(.failure(error))
+                break
```
Comment (at-truongnguyen) [inline MyApp/ViewModel/List/DrinkViewModel.swift]: haha `break`

File: MyApp/ViewModel/List/FavoriteViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    func getProductDetail(_ completion: @escaping GetProductCompletion) {
+        Api.Item.query { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success(let product):
+                this.items = product as? [Item]
+                completion(.success)
+            case .failure(let error):
+                completion(.failure(error))
+                break
```
Comment (at-truongnguyen) [inline MyApp/ViewModel/List/FavoriteViewModel.swift]: `break `

File: MyApp/ViewModel/List/FoodViewModel.swift [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    func getProductDetail(_ completion: @escaping GetProductCompletion) {
+        Api.Item.query { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success(let product):
+                this.items = product as? [Item]
+                completion(.success)
+            case .failure(let error):
+                completion(.failure(error))
+                break
```
Comment (at-truongnguyen) [inline MyApp/ViewModel/List/FoodViewModel.swift]: `break`

File: MyApp/View/Controllers/List/Cell/ItemListTableCell.swift [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        // Initialization code
+    }
+
+    func updateView() {
+        guard let viewModel = viewModel else { return }
+        itemBackgroundImage.image = UIImage(named: "\(viewModel.imageName)")
+        itemNameLabel.text = viewModel.name
+        if viewModel.detail.isEmpty {
+            itemDetailLabel.text = App.Strings.defaultItemDetail
+        }
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/List/Cell/ItemListTableCell.swift]: Làm đoạn lệnh ni trong viewModel em ơi :)

**Pull request merged**